src/plugins/pcr_primer_design_for_dna_assembly/src/tasks/PCRPrimerDesignForDNAAssemblyTaskTest.cpp

```diff
@@ -0,0 +1,127 @@
+#define OTHER_SEQ_PCR_URL_ATTR "otherSequencesInPcrUrl"
+
+void GTest_PCRPrimerDesignForDNAAssemblyTaskTest::init(XMLTestFormat *, const QDomElement &el) {
+    docName = el.attribute(DOC_NAME_ATTR);
+    seqName = el.attribute(SEQ_NAME_ATTR);
+    settings.gibbsFreeEnergy = getU2RangeInt(el, ENERGY_ATTR, ";");
+    settings.meltingPoint = getU2RangeInt(el, MELTING_POINT_ATTR, ";");
+    settings.overlapLength = getU2RangeInt(el, OVERLAP_ATTR, ";");
+    settings.gibbsFreeEnergyExclude = getInt(el, ENERGY_EXCLUDE_ATTR);
+    settings.meltingPointExclude = getInt(el, MELTING_POINT_EXCLUDE_ATTR);
+    settings.complementLengthExclude = getInt(el, COMPL_LENGTH_EXCLUDE_ATTR);
+    QString buf = el.attribute(INSERT_TO_ATTR);
+    if (buf == "5") {
+        settings.insertTo = PCRPrimerDesignForDNAAssemblyTaskSettings::BackboneBearings::Backbone5;
+    } else if (buf == "3") {
+        settings.insertTo = PCRPrimerDesignForDNAAssemblyTaskSettings::BackboneBearings::Backbone3;
+    } else {
+        wrongValue(INSERT_TO_ATTR);
+        return;
+    }
+    settings.leftArea = getU2Region(el, LEFT_AREA_ATTR);
+    settings.rightArea = getU2Region(el, RIGHT_AREA_ATTR);
+    result = getU2RegionList(el, RESULT_ATTR, ";", 8);
+    buf = el.attribute(BACKBONE_URL_ATTR);
+    if (!buf.isEmpty()) {
+        QFileInfo refFile(env->getVar("COMMON_DATA_DIR") + "/" + buf);
+        if (!refFile.exists()) {
+            stateInfo.setError(QString("file not exist %1").arg(refFile.absoluteFilePath()));
+            return;
+        }
+        settings.backboneSequenceUrl = env->getVar("COMMON_DATA_DIR") + "/" + buf;
+    }
+    buf = el.attribute(OTHER_SEQ_PCR_URL_ATTR);
+    if (!buf.isEmpty()) {
+        QFileInfo refFile(env->getVar("COMMON_DATA_DIR") + "/" + buf);
+        if (!refFile.exists()) {
+            stateInfo.setError(QString("file not exist %1").arg(refFile.absoluteFilePath()));
+            return;
+        }
+        settings.otherSequencesInPcrUrl = env->getVar("COMMON_DATA_DIR") + "/" + buf;
+    }
+}
+
+void GTest_PCRPrimerDesignForDNAAssemblyTaskTest::prepare() {
+    Document *loadedDocument = getContext<Document>(this, docName);
+    if (loadedDocument == nullptr) {
+        stateInfo.setError(GTest::tr("context not found %1").arg(docName));
+        return;
+    }
+    U2SequenceObject *dnaso = (U2SequenceObject *)loadedDocument->findGObjectByName(seqName);
+    if (dnaso == nullptr) {
+        stateInfo.setError(GTest::tr("Sequence %1 not found").arg(seqName));
+        return;
+    }
+    task = new PCRPrimerDesignForDNAAssemblyTask(settings, dnaso->getWholeSequenceData(stateInfo));
+    CHECK_OP(stateInfo, );
+    addSubTask(task);
+}
+
+Task::ReportResult GTest_PCRPrimerDesignForDNAAssemblyTaskTest::report() {
+    const QList<U2Region> actualResults = task->getResults();
+    if (actualResults != result) {
+        CHECK_EXT(actualResults.size() == 8, setError(QString("Actual results size %1, expected 8.")
```

Comment:
It could be less than eight results, is that handled?

Reply:
We know how this task works, and we know exact number of results.